framework/include/base/MooseFunctor.h

```diff
@@ -28,115 +28,115 @@ class FaceInfo;
 
 namespace Moose
 {
-/**
- * A non-templated base class for functors that allow an owner object to hold
- * different class template instantiations of \p Functor in a single container
- */
-class FunctorBase
+struct ElemArg
 {
-public:
-  FunctorBase() = default;
-  virtual ~FunctorBase() = default;
+  const libMesh::Elem * elem;
+  bool correct_skewness;
+  bool apply_gradient_to_skewness;
+};
 
-  ///@{
-  /**
-   * Virtual methods meant to be used for handling functor evaluation cache clearance
-   */
-  virtual void timestepSetup() = 0;
-  virtual void residualSetup() = 0;
-  virtual void jacobianSetup() = 0;
-  ///@}
+struct ElemFromFaceArg
+{
+  const libMesh::Elem * elem;
+  const FaceInfo * fi;
+  bool correct_skewness;
+  bool apply_gradient_to_skewness;
```

Comment:
exactly. I did not think of this on my own. I only created this member because your current implementation requires it. I think this is something that you've said we can refactor/remove in the future?

Reply:
Yes, I am planning to touch this in the near future!